client/js/_player/components/common/clix_drag_and_drop/clix_drag_and_drop.jsx

```diff
@@ -0,0 +1,220 @@
+import React  from 'react';
+import _      from 'lodash';
+
+import withDragDropContext  from '../with_drag_drop_context';
+import ClixDropZone         from './clix_drop_zone';
+import Droppable            from './droppable'; // eslint-disable-line import/no-named-as-default
+import CustomDragLayer      from '../custom_drag_layer'; // eslint-disable-line import/no-named-as-default
+
+export class ClixDragAndDrop extends React.Component {
+  static propTypes = {
+    answers: React.PropTypes.arrayOf(React.PropTypes.shape({
+      id: React.PropTypes.string, // eslint-disable-line react/no-unused-prop-types
+      text: React.PropTypes.string, // eslint-disable-line react/no-unused-prop-types
+    })),
+    zones: React.PropTypes.arrayOf(React.PropTypes.shape({
+      spatialUnit: React.PropTypes.shape({ // eslint-disable-line react/no-unused-prop-types
+        coordinateValues: React.PropTypes.arrayOf( // eslint-disable-line react/no-unused-prop-types
+          React.PropTypes.oneOfType([React.PropTypes.string, React.PropTypes.number])
+        ),
+        width: React.PropTypes.oneOfType( // eslint-disable-line react/no-unused-prop-types
+          [React.PropTypes.string, React.PropTypes.number]
+        ),
+        height: React.PropTypes.oneOfType( // eslint-disable-line react/no-unused-prop-types
+          [React.PropTypes.string, React.PropTypes.number]
+        )
+      }),
+      dropBehaviorType: React.PropTypes.string, // eslint-disable-line react/no-unused-prop-types
+    })),
+    targets: React.PropTypes.arrayOf(React.PropTypes.shape({
+      text: React.PropTypes.string, // eslint-disable-line react/no-unused-prop-types
+    })),
+    selectAnswer: React.PropTypes.func,
+    selectedAnswers: React.PropTypes.arrayOf(React.PropTypes.shape({
+      id: React.PropTypes.shape({
+        id: React.PropTypes.string, // eslint-disable-line react/no-unused-prop-types
+        zoneIndex: React.PropTypes.number, // eslint-disable-line react/no-unused-prop-types
+      }),
+      droppable: React.PropTypes.shape({}),
+      containerId: React.PropTypes.number, // eslint-disable-line react/no-unused-prop-types
+      coordinateValues: React.PropTypes.arrayOf( // eslint-disable-line react/no-unused-prop-types
+        React.PropTypes.oneOfType([React.PropTypes.string, React.PropTypes.number])
+      )
+    }))
+  };
+
+  deselectAnswer(item) { // eslint-disable-line react/sort-comp
```

Comment:
Rip the prop types out, and move the functions. Our linter should not junk up our code (with exception). We need to either change the rule (and in the case of prop-types it is more valuable to have the rule than not) or change the code. This just can't become a pattern in our code. I am willing to re-evaluate the function order rule at some point (but not his PR).